salt/modules/boto_elasticsearch_domain.py

```diff
@@ -0,0 +1,469 @@
+
+# Import Python libs
+from __future__ import absolute_import
+import logging
+import json
+from distutils.version import LooseVersion as _LooseVersion  # pylint: disable=import-error,no-name-in-module
+
+# Import Salt libs
+import salt.utils.boto3
+import salt.utils.compat
+import salt.utils
+from salt.exceptions import SaltInvocationError
+from salt.ext.six import string_types
+
+log = logging.getLogger(__name__)
+
+# Import third party libs
+
+# pylint: disable=import-error
+try:
+    #pylint: disable=unused-import
+    import boto
+    import boto3
+    #pylint: enable=unused-import
+    from botocore.exceptions import ClientError
+    logging.getLogger('boto').setLevel(logging.CRITICAL)
+    logging.getLogger('boto3').setLevel(logging.CRITICAL)
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+# pylint: enable=import-error
+
+
+def __virtual__():
+    '''
+    Only load if boto libraries exist and if boto libraries are greater than
+    a given version.
+    '''
+    required_boto_version = '2.8.0'
+    required_boto3_version = '1.2.5'
+    # the boto_lambda execution module relies on the connect_to_region() method
+    # which was added in boto 2.8.0
+    # https://github.com/boto/boto/commit/33ac26b416fbb48a60602542b4ce15dcc7029f12
+    if not HAS_BOTO:
+        return (False, 'The boto_lambda module could not be loaded: '
+                'boto libraries not found')
+    elif _LooseVersion(boto.__version__) < _LooseVersion(required_boto_version):
+        return (False, 'The boto_lambda module could not be loaded: '
+                'boto version {0} or later must be installed.'.format(required_boto_version))
+    elif _LooseVersion(boto3.__version__) < _LooseVersion(required_boto3_version):
+        return (False, 'The boto_lambda module could not be loaded: '
+                'boto version {0} or later must be installed.'.format(required_boto3_version))
+    else:
+        return True
+
+
+def __init__(opts):
+    salt.utils.compat.pack_dunder(__name__)
+    if HAS_BOTO:
+        __utils__['boto3.assign_funcs'](__name__, 'es')
+
+
+def exists(DomainName,
+           region=None, key=None, keyid=None, profile=None):
+    '''
+    Given a domain name, check to see if the given domain exists.
+
+    Returns True if the given domain exists and returns False if the given
+    function does not exist.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_elasticsearch_domain.exists mydomain
+
+    '''
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    try:
+        domain = conn.describe_elasticsearch_domain(DomainName=DomainName)
+        return {'exists': True}
+    except ClientError as e:
+        if e.response.get('Error', {}).get('Code') == 'ResourceNotFoundException':
+            return {'exists': False}
+        return {'error': salt.utils.boto3.get_error(e)}
+
+
+def status(DomainName,
+             region=None, key=None, keyid=None, profile=None):
```

Comment:
Same comment here about `DomainName` to `domain_name`. Can you also line these kwargs up with the line above?
```
def status(domain_name,
                 region=None, key=None, keyid=None, profile=None)
```